compiler/rustc_middle/src/ty/context.rs

```diff
@@ -367,6 +368,11 @@ pub struct GeneratorInteriorTypeCause<'tcx> {
     pub expr: Option<hir::HirId>,
 }
 
+#[derive(TyEncodable, TyDecodable, Clone, Debug, Eq, PartialEq, HashStable)]
+pub struct RvalueScopes {
+    pub inner: FxHashMap<hir::ItemLocalId, Option<Scope>>,
```

Comment:
```suggestion
    map: FxHashMap<hir::ItemLocalId, Option<Scope>>,
```

Nit: I'd move the things that manipulate the map to methods on `RvalueScopes` (and move this definition over to where its `impl` is)

Reply:
Yes, I think making it `pub` was actually not necessary.